src/applications/personalization/profile-2/components/MobileMenuTrigger.jsx

```diff
@@ -0,0 +1,133 @@
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+import {
+  closeSideNav as closeSideNavAction,
+  openSideNav as openSideNavAction,
+  pinMenuTrigger as pinMenuTriggerAction,
+  unpinMenuTrigger as unpinMenuTriggerAction,
+} from '../actions';
+
+const MobileMenuTrigger = ({
+  closeSideNav,
+  focusTriggerButton,
+  openSideNav,
+  isMenuTriggerPinned,
+  pinMenuTrigger,
+  unpinMenuTrigger,
+}) => {
+  const button = useRef(null);
+  const placeholder = useRef(null);
+
+  const [mobileBreakpoint] = useState(767);
+  const [isMobile, setIsMobile] = useState(false);
+  const [triggerPosition, setTriggerPosition] = useState(null);
+  const [triggerHeight, setTriggerHeight] = useState(null);
+
+  // init some local state
+  useEffect(
+    () => {
+      setTriggerHeight(window.getComputedStyle(button.current).height);
+      setTriggerPosition(placeholder.current.offsetTop);
+      setIsMobile(window.innerWidth <= mobileBreakpoint);
+    },
+    [mobileBreakpoint],
+  );
+
+  useEffect(() => {
+    const handleResize = () => {
+      const mobile = window.innerWidth <= mobileBreakpoint;
+      if (!mobile) {
+        if (isMobile) {
+          setIsMobile(false);
+          closeSideNav();
+        }
+        return;
+      }
+      // The position of the trigger might move up or down as the window width
+      // changes and elements above it compress and take up more vertical space
+      // so we need to reset it on window resize
+      setTriggerPosition(placeholder.current.offsetTop);
+      if (!isMobile) {
+        setIsMobile(true);
+      }
+    };
+    // pin or unpin the mobile trigger
+    const handleScroll = () => {
+      if (!isMobile) return;
+      if (window.pageYOffset >= triggerPosition && !isMenuTriggerPinned) {
+        pinMenuTrigger();
+        placeholder.current.style.height = triggerHeight;
+      }
+      if (window.pageYOffset < triggerPosition && isMenuTriggerPinned) {
+        unpinMenuTrigger();
+        placeholder.current.style.height = 0;
+      }
+    };
+
+    window.addEventListener('resize', handleResize);
+    window.onscroll = handleScroll;
+
+    if (focusTriggerButton) {
+      button.current.focus();
+    }
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+      window.onscroll = null;
+    };
+  });
+
+  const buttonClasses = classnames({ fixed: isMenuTriggerPinned });
+
+  return (
+    <div className="va-btn-sidebarnav-trigger">
+      {/*
+      This invisible placeholder fills the vertical space normally taken up by the `button` when the `button`'s position is fixed and it is pulled out of the normal page flow.
+      */}
+      <div ref={placeholder} />
+      <button
+        type="button"
+        className={buttonClasses}
+        aria-controls="va-detailpage-sidebar"
+        onClick={openSideNav}
+        ref={button}
+      >
+        <span>
+          <b>Profile Menu</b>
+          <svg
```

Comment:
Can we use an `img` tag and pass the `svg` to the `src` prop? That would be similar to what's happening in `src/applications/claims-status/components/appeals-v2/DocketCard.jsx`?

Something like:

```
import mobileMenuSvg from '../path'

<img src={mobileMenuSvg} />
```

I'm also OK with you keeping it as is, just seems a bit cleaner :)

Reply:
sadly we are not set up to import SVGs like like, but we can switch to using a `img` instead. Thanks for finding the DocketCard example! I wasn't sure where I should save the white arrow SVG that we are defining inline, but I found that we already have one! https://github.com/department-of-veterans-affairs/vets-website/blob/f841bc4b8b595b84870ab3d21b46464fceb434f5/src/applications/personalization/profile-2/components/MobileMenuTrigger.jsx#L1 woohoo!

Reply:
nailed it in my first attempt 
![image](https://user-images.githubusercontent.com/20728956/80501826-bfec8c00-8924-11ea-9d03-3936dadbc893.png)